openedx/core/djangoapps/content_libraries/docs/decisions/0002-libraries-lti-provider.rst

```diff
@@ -0,0 +1,30 @@
+Allow content libraries to used by LTI consumers
+------------------------------------------------
+
+Status
+------
+
+Pending
+
+Context
+-------
+
+Currently, there's no way for xblocks in blockstore-based content libraries to be served to other LTI consumers. This ADR explores ways in which this can be achieved, without overly complicating the codebase. The platform currently does have an :code:`lms.lti_provider` app which implements the LTI 1.1 spec, but it only works for modulestore-based courses, and the LTI 1.1 spec itself is also deprecated and deemed insecure.
+
+Decision
+--------
+
+We will use the latest LTI spec (`LTI 1.3`_) to serve the xblocks. This would be largely independent of the existing LTI 1.1 implementation, and would barely require any changes in that app.
+
+Instead, we would add an :code:`lti` package to the :code:`openedx.core.content_libraries` app, which will handle LTI endpoints, and communicate with the LTI consumers. It would be a good idea to use a package like `pylti1.3`_ which can handle the LTI-spec implementation, which saves a lot of development and testing effort, and keeps the platform codebase relevant to openedx.
+
+.. _LTI 1.3: http://www.imsglobal.org/activity/learning-tools-interoperability
+.. _pylti1.3: https://github.com/dmitry-viskov/pylti1.3
+
+
+Consequences
+------------
+
+Once implemented, content from blockstore-based libraries can be easily embedded into other LMS platforms. This is made easier by the fact that these libraries are deisgned to be embeddable, and can be easily rendered inside iframes.
+
+The rest of the functionality would be similar to how the current LTI provider works, having similar LTI login, setup, and grade syncing.
```

Comment:
I would add information to the effect that the new endpoints will be libraries-v2 specific.